chaco/legend.py

```diff
@@ -355,7 +358,7 @@ def get_preferred_size(self):
         if len(self.plots) == 0:
             return [0, 0]
 
-        plot_names, visible_plots = map(list, zip(*sorted(self.plots.items())))
+        plot_names, visible_plots = list(sm.map(list, sm.zip(*sorted(six.iteritems(self.plots)))))
```

Comment:
Ah, I misunderstood. Thanks for the clarification, and let's keep it like you had it originally.